dev/tools/update_icons.dart

```diff
@@ -348,24 +346,24 @@ class _Icon {
 
     flutterId = id;
     for (final MapEntry<String, String> rewritePair in identifierRewrites.entries) {
-      if (id.startsWith(rewritePair.key)) {
-        flutterId = id.replaceFirst(rewritePair.key, identifierRewrites[rewritePair.key]);
+      if (id!.startsWith(rewritePair.key)) {
+        flutterId = id!.replaceFirst(rewritePair.key, identifierRewrites[rewritePair.key]!);
       }
     }
 
-    name = id.replaceAll('_', ' ');
+    name = id!.replaceAll('_', ' ');
   }
 
   static const List<String> styleSuffixes = <String>['', '_outlined', '_rounded', '_sharp'];
 
-  String id;            // e.g. 5g, 5g_outlined, 5g_rounded, 5g_sharp
-  String shortId;       // e.g. 5g
-  String flutterId;     // e.g. five_g, five_g_outlined, five_g_rounded, five_g_sharp
-  String name;          // e.g. five g, five g outlined, five g rounded, five g sharp
-  String hexCodepoint;  // e.g. e547
+  String? id;            // e.g. 5g, 5g_outlined, 5g_rounded, 5g_sharp
+  String? shortId;       // e.g. 5g
+  String? flutterId;     // e.g. five_g, five_g_outlined, five_g_rounded, five_g_sharp
+  String? name;          // e.g. five g, five g outlined, five g rounded, five g sharp
+  String? hexCodepoint;  // e.g. e547
```

Comment:
Why not using "late" keyword for these properties since they will be initialized in the constructor?

Reply:
Late should be used only in tests.

Reply:
`late` can also be used in non-test code. I think if any of these properties are going to be defined no matter what, `late` would be better than a nullable `String`. That way, we also do not have to use `!` all over the code, since it seems to appear quite a bit.

Reply:
> `late` can also be used in non-test code. I think if any of these properties are going to be defined no matter what, `late` would be better than a nullable `String`. That way, we also do not have to use `!` all over the code, since it seems to appear quite a bit.

Agreed.